nomenclature/config.py

```diff
@@ -11,29 +12,94 @@
     field_validator,
     model_validator,
     ConfigDict,
-    BeforeValidator,
 )
+from nomenclature.code import Code
+from pyam.str import escape_regexp
+
+
+class RepositoryWithFilter(BaseModel):
```

Comment:
The "with-filter" seems important now because of the changes, but it isn't really the defining characteristic of this class - and there is no **RepositoryWithoutFilter** class.
```suggestion
class CodeListFromRepository(BaseModel):
```